ff/src/fields/arithmetic.rs

```diff
@@ -5,12 +5,26 @@
 /// `P::MODULUS` has (a) a non-zero MSB, and (b) at least one
 /// zero bit in the rest of the modulus.
 macro_rules! impl_field_mul_assign {
+    (1) => {
+        fn mul_assign(&mut self, other: &Self) {
+            #[inline(always)]
+            fn mont_reduce(&mut self, mul_result: u128) {
```

Comment:
This is just a specialization of the standard algorithm to the case where there is just one limb.